modules/standard/HDFS.chpl

```diff
@@ -18,19 +18,315 @@
+----------------------------
+
+There are three ways provided to open HDFS files within Chapel.
+
+Using an HDFS filesystem with open(url="hdfs://...")
+****************************************************
+
+.. code-block:: chapel
+
+  // Open a file on HDFS connecting to the default HDFS instance
+  var f = open(mode=iomode.r, url="hdfs://host:port/path");
+
+  // Open up a reader and read from the file
+  var reader = f.reader();
+
+  // ...
+
+  reader.close();
+
+  f.close();
+
+
+Explicitly Using Replicated HDFS Filesystems and Files
+******************************************************
+
+.. code-block:: chapel
+
+  use HDFS;
+
+  // Connect to HDFS via the default username (or whichever you want)
+  //
+  var hdfs = hdfsChapelConnect("default", 0);
+
+  //
+  // Create a file per locale
+  //
+  var gfl  = hdfs.hdfsOpen("/user/johnDoe/isThisAfile.txt", iomode.r);
+
+  ...
+  //
+  // On any given locale, you can get the local file for the locale that
+  // the task is currently running on via:
+  //
+  var fl = gfl.getLocal();
+
+  // This file can be used as with a traditional file in Chapel, by
+  // creating reader channels on it.
+
+  // When you are done and want to close the files and disconnect from
+  // HDFS, use:
+
+  gfl.hdfsClose();
+  hdfs.hdfsChapelDisconnect();
+
+Explicitly Using Local HDFS Filesystems and Files
+*************************************************
+
+The HDFS module file also supports non-replicated values across
+locales. So if you only wanted to connect to HDFS and open a file on
+locale 1 you could do:
+
+.. code-block:: chapel
+
+  on Locales[1] {
+    var hdfs = hdfs_chapel_connect("default", 0);
+    var fl = hdfs.hdfs_chapel_open("/user/johnDoe/myFile.txt", iomode.cw);
+    ...
+    var read = fl.reader();
+    ...
+    fl.close();
+    hdfs.hdfs_chapel_disconnect();
+  }
+
+The only stipulations are that you cannot open a file in both read and
+write mode at the same time. (i.e iomode.r and iomode.cw are the only
+modes that are supported, due to HDFS limitations).
+
+
+.. _setting-up-hadoop:
+
+Setting up Hadoop 
+-----------------
+
+If you have a working installation of Hadoop already, you can skip
+this section, other than to set up your CLASSPATH environment
+variable.  This section is written so that people without sudo
+permission can install and use HDFS.  If you do have sudo permissions,
+you can usually install all of these via a package manager.
+
+The general outline for these instructions is: 
+  1. Install and point to the jdk to provide code Chapel needs to
+     compile against libhdfs
+  2. Install Hadoop
+  3. Set up Hadoop on (a) the local host or (b) a cluster of hosts
+  4. Start up HDFS
+  5. Stop HDFS when you're done
+  6. Set up Chapel to run in distributed mode
+
+First reflect your directory structure and version numbers (etc) in
+the code at the bottom of this page and put it in your .bashrc (or
+.bash_profile -- your choice) and source whichever one you put it
+into.
```

Comment:
Since the sample .bashrc is followed by some code definitions, we may want to link to it instead here.